File: SpendingTracker-Mobile/app/src/main/java/vn/edu/fpt/spendingtracker_mobile/dtos/PagedListResult.java
package vn.edu.fpt.spendingtracker_mobile.dtos;

import java.math.BigDecimal;
import java.util.List;

public class PagedListResult<TItemType> {
    public BigDecimal totalBudget;
    public int totalItemCount;
    public int pageCount;
    public int pageSize;
    public int pageNumber;
    public List<TItemType> items;

    public PagedListResult(BigDecimal totalBudget, int totalItemCount, int pageCount, int pageSize, int pageNumber, List<TItemType> items) {
        this.totalBudget = totalBudget;
        this.totalItemCount = totalItemCount;
        this.pageCount = pageCount;
        this.pageSize = pageSize;
        this.pageNumber = pageNumber;
        this.items = items;
    }

    public PagedListResult(int totalItemCount, int pageCount, int pageSize, int pageNumber, List<TItemType> items) {
        this.totalItemCount = totalItemCount;
        this.pageCount = pageCount;
        this.pageSize = pageSize;
        this.pageNumber = pageNumber;
        this.items = items;
    }
}

File: SpendingTracker-API/DTOs/Web_Mobile/PagedListResult.cs
namespace SpendingTracker_API.DTOs.Web_Mobile
{
    public class PagedListResult<TItemType> where TItemType : class
    {
        public decimal? TotalBudget { get; set; }
        public int TotalItemCount { get; set; }
        public int PageCount { get; set; }
        public int PageSize { get; set; }
        public int PageNumber { get; set; }
        public IEnumerable<TItemType> Items { get; set; }
    }
}
